lbrynet/lbrynet_daemon/auth/client.py

```diff
@@ -0,0 +1,168 @@
+        self.error = rpc_error
+
+
+class LBRYAPIClient(object):
+    __api_token = None
+
+    def __init__(self, key_name=None, key=None, pw_path=None, timeout=HTTP_TIMEOUT, connection=None, count=0,
+                                            service=None, cookies=None, auth=None, url=None, login_url=None):
+        self.__api_key_name = API_KEY_NAME if not key_name else key_name
+        self.__api_token = key
+        self.__pw_path = os.path.join(DATA_DIR, ".api_keys") if not pw_path else pw_path
+        self.__service_name = service
+
+        if not key:
+            keys = load_api_keys(self.__pw_path)
+            api_key = keys.get(self.__api_key_name, False)
+            self.__api_token = api_key['token']
+            self.__api_key_obj = api_key
+        else:
+            self.__api_key_obj = APIKey({'token': key})
+
+        if login_url is None:
+            self.__service_url = "http://%s:%s@%s:%i/%s" % (self.__api_key_name, self.__api_token,
+                                                               API_INTERFACE, API_PORT, API_ADDRESS)
+        else:
+            self.__service_url = login_url
+
+        self.__id_count = count
+
+        if auth is None and connection is None and cookies is None and url is None:
+            self.__url = urlparse.urlparse(self.__service_url)
+            (user, passwd) = (self.__url.username, self.__url.password)
+            try:
+                user = user.encode('utf8')
+            except AttributeError:
+                pass
+            try:
+                passwd = passwd.encode('utf8')
+            except AttributeError:
+                pass
+            authpair = user + b':' + passwd
+            self.__auth_header = b'Basic ' + base64.b64encode(authpair)
+
+            self.__conn = requests.Session()
+            self.__conn.auth = (user, passwd)
+
+            req = requests.Request(method='POST',
+                                   url=self.__service_url,
+                                   auth=self.__conn.auth,
+                                   headers={'Host': self.__url.hostname,
+                                            'User-Agent': USER_AGENT,
+                                            'Authorization': self.__auth_header,
+                                            'Content-type': 'application/json'},)
+            r = req.prepare()
+            http_response = self.__conn.send(r)
+            cookies = http_response.cookies
+            self.__cookies = cookies
+            # print "Logged in"
+
+            uid = cookies.get('TWISTED_SESSION')
+            api_key = APIKey.new(seed=uid)
+            # print "Created temporary api key"
+
+            self.__api_token = api_key.token()
+            self.__api_key_obj = api_key
+        else:
+            self.__auth_header = auth
+            self.__conn = connection
+            self.__cookies = cookies
+            self.__url = url
+
+            if cookies.get("secret", False):
```

Comment:
cookie name should be a constant, not hardcoded. also, the name should probably start with "lbry_", so its easier to debug in the browser later